ansible/roles/providers/tasks/qemu.yml

```diff
@@ -62,8 +62,17 @@
   when: ansible_os_family == "RedHat"
 
 - name: Disable Hyper-V KVP protocol daemon on Ubuntu
-  systemd:
-    name: hv-kvp-daemon
-    state: stopped
-    enabled: false
+  block:
+    - name: Attempt to stop the service
+      service:
+        name:  hv-kvp-daemon
+        state: stopped
+  rescue:
+    - name: Get the list of services
+      service_facts:
+
+    - name: Verify that hv-kvp-daemon is not installed
+      assert:
+        that:
+          - "'hv-kvp-daemon.service' not in services"
```

Comment:
I think the intent is to stop and disable the service if and only if it is installed. Can we do that without the complexity of a `rescue`?

For example, can we always populate `service_facts`, then stop and disable if `hv-kvp-daemon` is in the list of services?

Reply:
Absolutely. Good idea, I will amend the commit and re-submit.

Reply:
Made your suggested change, apologies for the bunch of commits - I messed up a merge due to weak git-fu ;)

I can rebase and squash if you'd prefer a single commit.

Reply:
Thanks for rebasing/squashing!